toolkit/tools/imagegen/installutils/installutils.go

```
@@ -1523,6 +1535,54 @@ func setGrubCfgIMA(grubPath string, kernelCommandline configuration.KernelComman
+		verityHashArg           = fmt.Sprintf("rd.verityroot.hashtree=/%s.hashtree", readOnlyRoot.MappedName)
+		verityRootHashArg       = fmt.Sprintf("rd.verityroot.roothashfile=/%s.roothash", readOnlyRoot.MappedName)
+		verityRootHashSigArg    = fmt.Sprintf("rd.verityroot.roothashsig=/%s.p7", readOnlyRoot.MappedName)
+		verityFECDataArg        = fmt.Sprintf("rd.verityroot.fecdata=/%s.fec", readOnlyRoot.MappedName)
+		verityFECRootsArg       = fmt.Sprintf("rd.verityroot.fecroots=%d", readOnlyRoot.FecRoots)
+		verityErrorHandling     = fmt.Sprintf("rd.verityroot.verityerrorhandling=%s", readOnlyRoot.ErrorBehavior)
+		verityValidateOnBootArg = fmt.Sprintf("rd.verityroot.validateonboot=%v", readOnlyRoot.ValidateOnBoot)
+		verityOverlaysArg       = fmt.Sprintf("rd.verityroot.overlays=\"%s\"", strings.Join(readOnlyRoot.TmpfsOverlays, " "))
+		verityDebugMountsArg    = fmt.Sprintf("rd.verityroot.overlays_debug_mount=%s", readOnlyRoot.TmpfsOverlaysDebugMount)
+		verityPattern           = "{{.ReadOnlyVerityRoot}}"
+		verityArgs              = ""
+
+		cmdline configuration.KernelCommandLine
+	)
+
+	if readOnlyRoot.MappedName != "" {
+		// Basic set of verity arguments common to all use cases
+		verityArgs = fmt.Sprintf("%s %s %s %s %s %s %s",
+			verityMountArg,
+			verityHashArg,
+			verityRootHashArg,
+			verityErrorHandling,
+			verityValidateOnBootArg,
+			verityOverlaysArg,
+			verityDebugMountsArg,
+		)
+		// Only include the FEC arguments if we have FEC enabled
+		if readOnlyRoot.FecRoots > 0 {
+			verityArgs = fmt.Sprintf("%s %s %s", verityArgs, verityFECDataArg, verityFECRootsArg)
+		}
+		if readOnlyRoot.UseRootHashSignature {
+			verityArgs = fmt.Sprintf("%s %s", verityArgs, verityRootHashSigArg)
+		}
+	}
+
+	logger.Log.Debugf("Adding Verity Root ('%s') to %s", verityArgs, grubPath)
+	err = sed(verityPattern, verityArgs, cmdline.GetSedDelimeter(), grubPath)
+	if err != nil {
+		logger.Log.Warnf("Failed to set grub.cfg's IMA setting: %v", err)
```

Shouldn't this force a build break?

---

It should bubble the error out, this is the pattern we use on the other config update functions.

---
In reply to: [552717537](https://github.com/microsoft/CBL-Mariner/pull/483#discussion_r552717537) [](ancestors = 552717537)